langlib/langlib-test/src/test/java/org/ballerinalang/langlib/test/LangLibArrayTest.java

```diff
@@ -461,9 +461,37 @@ public void testArrayLibNegativeCases() {
                         "found 'function[]'", 199, 15);
         BAssertUtil.validateError(negativeResult, errorIndex++, "incompatible types: expected " +
                 "'ballerina/lang.array:0.0.0:AnydataType', found 'function (int) returns (int)'", 199, 26);
+        BAssertUtil.validateError(negativeResult, errorIndex++,
+                "undefined function 'som' in type '[int,int]'", 207, 16);
+        BAssertUtil.validateError(negativeResult, errorIndex++,
+                "incompatible types: expected 'function (ballerina/lang.array:0.0.0:Type) returns (boolean)', " +
+                        "found 'function (int) returns (int)'", 215, 21);
+        BAssertUtil.validateError(negativeResult, errorIndex++,
+                "incompatible types: expected 'int', found 'boolean'", 219, 13);
+        BAssertUtil.validateError(negativeResult, errorIndex++,
+                "incompatible types: expected 'int', found 'string'", 224, 18);
+        BAssertUtil.validateError(negativeResult, errorIndex++,
+                "operator '>' not defined for 'string' and 'int'", 229, 25);
         Assert.assertEquals(negativeResult.getErrorCount(), errorIndex);
     }
 
+    @Test
+    public void testArrayLibNegativeSemantics() {
+        CompileResult negativeResult = BCompileUtil.compile("test-src/arraylib_test_negative_semantic.bal");
+        int errorIndex = 0;
+        BAssertUtil.validateError(negativeResult, errorIndex++,
+                "incompatible types: 'int' will not be matched to 'string'", 18, 28);
+        Assert.assertEquals(negativeResult.getErrorCount(), errorIndex);
+    }
+
+    @Test(expectedExceptions = BLangRuntimeException.class,
+            expectedExceptionsMessageRegExp = "error: \\{ballerina/lang.array}IndexOutOfRange " +
+                    "\\{\"message\":\"array index out of range: index: 2, size: 2\"}.*")
```

Comment:
We can do these assertions in b7a itself. We can trap the error and validate the message and "message" field from the detail.

Reply:
We can get rid of `Assert.fail` also.

Reply:
Changed